File: MyAFBase/MyAFBaseTests/WeatherHeroThemeTests.swift
import Foundation
import Testing
@testable import MyAFBase

struct WeatherHeroThemeTests {
    @Test func clearUsesSunnyPalette() {
        let palette = WeatherHeroTheme.palette(for: "Clear")
        #expect(palette.symbol == "sun.max.fill")
        #expect(palette.colors.count == 3)
    }

    @Test func rainUsesRainPalette() {
        let palette = WeatherHeroTheme.palette(for: "rain")
        #expect(palette.symbol == "cloud.rain.fill")
    }

    @Test func thunderstormUsesStormPalette() {
        let palette = WeatherHeroTheme.palette(for: "thunderstorm")
        #expect(palette.symbol == "cloud.bolt.rain.fill")
    }

    @Test func unknownFallsBackToNeutralPalette() {
        let palette = WeatherHeroTheme.palette(for: "Unavailable")
        #expect(palette.symbol == "building.2.fill")
    }
}

File: android/app/src/test/java/com/ryanladuca/myafbase/ui/theme/WeatherHeroThemeTests.kt
package com.ryanladuca.myafbase.ui.theme

import org.junit.Assert.assertEquals
import org.junit.Test

class WeatherHeroThemeTests {
    @Test
    fun clearUsesSunnyPalette() {
        val palette = WeatherHeroTheme.palette("clear")
        assertEquals(3, palette.colors.size)
        assertEquals(0.18f, palette.symbolOpacity, 0.001f)
    }

    @Test
    fun rainUsesRainPalette() {
        val palette = WeatherHeroTheme.palette("rain")
        assertEquals(3, palette.colors.size)
        assertEquals(0.15f, palette.symbolOpacity, 0.001f)
    }

    @Test
    fun thunderstormUsesStormPalette() {
        val palette = WeatherHeroTheme.palette("thunderstorm")
        assertEquals(3, palette.colors.size)
        assertEquals(0.16f, palette.symbolOpacity, 0.001f)
    }

    @Test
    fun overcastThemeKeyUsesCloudyPalette() {
        val palette = WeatherHeroTheme.palette("cloudy")
        assertEquals(3, palette.colors.size)
        assertEquals(0.14f, palette.symbolOpacity, 0.001f)
    }

    @Test
    fun unknownFallsBackToNeutralPalette() {
        val palette = WeatherHeroTheme.palette("unavailable")
        assertEquals(2, palette.colors.size)
        assertEquals(0.10f, palette.symbolOpacity, 0.001f)
    }
}
